kotlinx-coroutines-core/common/src/flow/operators/Lint.kt

```diff
@@ -186,3 +186,15 @@ public suspend inline fun <T> SharedFlow<T>.toSet(destination: MutableSet<T>): N
 @InlineOnly
 public suspend inline fun <T> SharedFlow<T>.count(): Int =
     (this as Flow<T>).count()
+
+/**
+ * @suppress
+ */
+@Suppress("DeprecatedCallableAddReplaceWith")
+@Deprecated(
+    level = DeprecationLevel.WARNING,
+    message = "SharedFlow never completes, so this terminal operation never completes. Use 'replayCache.last()' for reading the last emitted item in the replay cache.",
+    replaceWith = ReplaceWith("replayCache.last()")
+)
+@InlineOnly
+public fun <T> SharedFlow<T>.last(): T = noImpl()
```

Comment:
`noImpl()` does not go well with `DeprecationLevel.WARNING`. It's either `noImpl()` + `ERROR` (so that code doesn't work + doesn't compile) or a proper implementation + `WARNING` (so that code works + compiles). Here, `WARNING` is the better choice, given that some code may already call this without encountering any real problems, so the implementation should be fixed. `count()` just above is a good inspiration.